packages/components/basic-table/stories/basic-table.stories.js

```diff
@@ -0,0 +1,185 @@
+import { storiesOf, select, object, text } from '@open-wc/demoing-storybook';
+import { withKnobs } from '@storybook/addon-knobs';
+import '@tradeshift/elements.basic-table';
+
+storiesOf('ts-basic-table', module)
+	.addDecorator(withKnobs)
+	.add(
+		'default',
+		() => {
+			const colsData = object('cols', [
+				{
+					property: 'context',
+					size: 'large',
+					value: 'Context',
+					renderer: 'renderCard'
+				},
+				{
+					visibility: 'desktop-only',
+					property: 'type',
+					size: 'small',
+					value: 'Type',
+					display: 'left',
+					renderer: 'renderStatus'
+				},
+				{ visibility: 'desktop-only', property: 'participants', size: 'small', value: 'Participants' },
+				{
+					visibility: 'desktop-only',
+					property: 'newActivity',
+					value: 'New Activity',
+					renderer: 'renderMobile'
+				},
+				{
+					visibility: 'desktop-only',
+					property: 'lastActivity',
+					value: 'Last Activity',
+					size: 'medium',
+					display: 'right'
+				},
+				{
+					visibility: 'mobile-only',
+					property: 'mobileLastActivity',
+					size: 'small',
+					display: 'right',
+					renderer: 'renderUnread'
+				}
+			]);
+			const data = object('data', [
+				{
+					context: {
+						name: 'Invoice #32131',
+						description: 'Tradeshift',
+						mobileDescription: 'Tradeshift | 5 participants',
+						type: 'offer'
+					},
+					type: { status: 'ok', text: 'External' },
+					participants: 100,
+					newActivity:
+						'Every little bunny has a habit that is funny. It doesnt matter where he goes he always wrinkles up his nose.',
+					lastActivity: 'Yesterday, 15:01',
+					mobileLastActivity: { unread: 10, lastActivity: 'Yesterday, 15:01' },
+					id: 1
+				},
+				{
+					context: {
+						name: 'Purchase Request #1231',
+						description: 'MegaImage',
+						mobileDescription: 'MegaImage',
+						type: 'document'
+					},
+					type: { status: 'error', text: 'Private' },
+					participants: 100,
+					newActivity:
+						'Every little bunny has a habit that is funny. It doesnt matter where he goes he always wrinkles up his nose.',
+					lastActivity: '29 Jul, 15:01',
+					mobileLastActivity: { unread: 0, lastActivity: '29 Jul, 15:01' },
+					id: 2
+				},
+				{
+					context: {
+						name: 'GoodsReceipt #231',
+						description: 'CocaCola',
+						mobileDescription: '5 participants',
+						type: 'private'
+					},
+					type: { status: 'note', text: 'Internal' },
+					participants: 100,
+					newActivity:
+						'Every little bunny has a habit that is funny. It doesnt matter where he goes he always wrinkles up his nose.',
+					lastActivity: '2 Aug, 15:01',
+					mobileLastActivity: { lastActivity: '2 Aug, 15:01' },
+					id: 3
+				}
+			]);
+			const dir = select(
+				'dir',
+				{
+					ltr: 'ltr',
+					rtl: 'rtl'
+				},
+				'ltr'
+			);
+			const selectedIds = object('selectedIds', [1]);
+			// eslint-disable-next-line no-eval
+			const renderCard = eval(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make sure that this dynamic injection or execution of code is safe. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.ts.sv/coding_rules#rule_key=javascript%3AEval)

Reply:
@Tradeshift/application-security Hi, This eval is safe since it's used in the storybook and the user is providing the function. Can you please approve this PR to be able to merge it with sonarQube issues.

Reply:
Hi @esmailpour-hosein 
You wrote:
> This eval is safe since it's used in the storybook and the user is providing the function

If a user provides the function, it does not mean it is safe. We actually tend to see it as quite the opposite :)
What do you mean by "it's used in the storybook"? Could you please elaborate?

People treat `eval()` quite harshly these days, and for a good reason. Is there no other way?
Introducing it adds potential risk in that one day, the function passed in might be malicious or just unsafe.
Could you think of a way to make sure this is indeed safe? There's no 99% safety in this context, I'm afraid.

Reply:
>If a user provides the function, it does not mean it is safe. We actually tend to see it as quite the opposite :)
Sorry, bad explanation. Now I see I really need to improve my english :)) 

>What do you mean by "it's used in the storybook"? Could you please elaborate?
It's a playground for developers to checkout the UI components with different properties, events, ... . 
You can see the storybook [here](https://tradeshift.github.io/elements/?path=/story/ts-file-card--default&knob-state=download&knob-size=&knob-error-message=Sample error message!&knob-rtl=&knob-removable=&knob-file-object={"name":"fileNamaae.pdf","size":123456789})
As you can see there's a knob tab which has different inputs. In this case we need to let the developer to be able change the default render function from one textarea like those in the knobs tab to just see the changes in the top frame (not saved anywhere).
The eval running the provided function in the knobs input, so the value of that input either coming from our source code which is safe, or from the user(developer who want to use our UI components). 
So first it's not going to be in the Tradeshift platform (only a separate github page), second it's not coming from user input that was saved in db and showed to another user, it's the developer him/herself. I completely understand your concern and totally agree! If it was inside the component that is going to be used in the Tradeshift platform, I wouldn't approve it. I can assure you it's 100% safe.

Reply:
Ok, thanks for the elaborate answer 👍 
I get it. This is `elements` and none of this is to be part of the platform.
Feel free to add a `NOSONAR` comment on this line to skip it. Something like:
```
const renderCard = eval(  // NOSONAR <our reasons for trusting this use of 'eval()'>
```

Reply:
Thanks! 👍 
@radu-marius-nedelcu please add this comment with explanation 🙏

Reply:
Just one thing to point out here @esmailpour-hosein: Please make sure that you always tag the security team whenever you'd use `NOSONAR`. People have tried to be sneaky in the past and that would just lead to us disabling it - which we don't really want to :)

This was merely an FYI in case you were not familiar with how this works!

Reply:
@esmailpour-hosein @asafiniu is this working for you now?

Reply:
@radu-marius-nedelcu Thanks 🙏

Reply:
`NOSONAR` is a bit too long here.
Sonarqube isn't picking up on this comment and for that blocks the merge nonetheless (green build, red status due to no new issues found, with a few left behind still unresolved).
Try a shorter, one-lined `NOSONAR` comment to see if that fixes it.

Reply:
it's not about being too long, but being on the wrong line. `// NOSONAR` needs to be on the exact same line where the violation is reported, since it only affects that one line

Reply:
Yeah, but auto-formatting (eslint) pushes it down a line. Thought it might be due to its length.

Reply:
Seriously..